lib/goo/sparql/solutions_mapper.rb

```diff
@@ -157,35 +164,56 @@ def get_value_object(id, objects_new, object, list_attributes, predicate)
         end
 
         if list_attributes.include?(predicate)
-          # To handle attr that are lists
-          pre = if @klass_struct
-                  @models_by_id[id][predicate]
-                else
-                  @models_by_id[id].instance_variable_get("@#{predicate}")
-                end
-          if object.nil? && pre.nil?
-            object = []
-          elsif object.nil? && !pre.nil?
-            object = pre
-          elsif object
-            object = !pre ? [object] : (pre.dup << object)
-            object.uniq!
-          end
+          pre = @klass_struct ? @models_by_id[id][predicate] : @models_by_id[id].instance_variable_get("@#{predicate}")
+
+          object = []  if object.nil? && pre.nil?
```

Comment:
this condition is not the same as before 
a possible refactoring is the following

```ruby 
if object.nil?
  object = pre.nil? ? [] : pre
else
  object = pre.nil? ? [object] : (pre.dup << object)
  object.uniq!
end 
```

Reply:
```
          object = []  if object.nil? && pre.nil?

          object = pre if object.nil? && !pre.nil?

          object = pre.nil? ? [object] : (pre.dup << object)
          object.uniq
          
          # version @
          if object.nil? 
            object = pre.nil? ? [] : pre 
          else 
            object = pre.nil? ? [object] : (pre.dup << object) object.uniq! 
          end
```